pkg/corerp/backend/deployment/deploymentprocessor.go

```diff
@@ -287,17 +287,31 @@ func (dp *deploymentProcessor) Deploy(ctx context.Context, id resources.ID, rend
 
 		err := dp.deployOutputResource(ctx, id, rendererOutput, computedValues, &handlers.PutOptions{Resource: &outputResource, DependencyProperties: deployedOutputResourceProperties})
 		if err != nil {
-			return rpv1.DeploymentOutput{}, err
+			failed := rpv1.DeploymentStatus(rpv1.DeploymentStatusFailed)
+			outputResource := rpv1.OutputResource{
+				LocalID: outputResource.LocalID,
+				ID:      outputResource.ID,
+				Status:  &failed,
+			}
+			deployedOutputResources = append(deployedOutputResources, outputResource)
+
+			return rpv1.DeploymentOutput{
+				DeployedOutputResources: deployedOutputResources,
+				ComputedValues:          computedValues,
+				SecretValues:            rendererOutput.SecretValues,
+			}, err
 		}
 
 		if outputResource.ID.IsEmpty() {
 			return rpv1.DeploymentOutput{}, fmt.Errorf("output resource %q does not have an id. This is a bug in the handler", outputResource.LocalID)
 		}
 
 		// Build database resource - copy updated properties to Resource field
+		success := rpv1.DeploymentStatus(rpv1.DeploymentStatusSucceeded)
```

Comment:
This can also be inlined. Can we have any other statuses here other than **Succeeded**?